actions/userinfo.js

```diff
@@ -1,17 +1,218 @@
+  };
+}
+function loginRequest() {
+  return {
+    type: LOGIN_REQUEST
+  };
+}
+
+export function loginSuccess(profile) {
+  return {
+    type: LOGIN_SUCCESS,
+    profile
+  };
+}
+
+function loginError(error) {
+  return {
+    type: LOGIN_ERROR,
+    error
+  };
+}
+export function loginLocation(location) {
+  return {
+    type: 'LOGIN_LOCATION',
+    location
+  };
+}
+export function checkLogin(location) {
+  return (dispatch, getState) => {
+    const location = getState().userInfo.loginLocation;
+    // Add callback for lock's `authenticated` event
+    lock.on('authenticated', authResult => {
+      return new Promise((res, rej) => {
+        res(authResult);
+      })
+        .then(authResult => {
+          let expiresAt = JSON.stringify(36000 * 1000 + new Date().getTime());
+          localStorage.setItem('access_token', authResult.accessToken);
+          localStorage.setItem('id_token', authResult.idToken); // static method
+          localStorage.setItem('expires_at', expiresAt);
+          return authResult;
+        })
+        .then(authResult => {
+          return dispatch(getUserInfo());
+        });
+    });
+    // });
+    // Add callback for lock's `authorization_error` event
+    lock.on('authorization_error', error => dispatch(loginError(error)));
+  };
+}
+export function updateUser(payload, userId) {
+  const data = {
+    user_metadata: { ...payload }
+  };
+  return dispatch => {
+    return Axios.patch(`https://gww.auth0.com/api/v2/users/${userId}`, data, {
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: 'Bearer ' + localStorage.getItem('id_token')
+      }
+    }).then(resp => {
+      return {
+        email: resp.data.email,
+        user_id: resp.data.user_id,
+        user_metadata: resp.data.user_metadata
+      };
+    });
+  };
+}
+
+export function resetPassword(email) {
+  return dispatch => {
+    const payload = {
+      client_id: 'OIzT7gmLYcxbLZWGDz7LAsX6i2iCP2tc',
+      email: email,
+      connection: 'Username-Password-Authentication'
+    };
+    return Axios.post(
+      'https://gww.auth0.com/dbconnections/change_password',
+      payload,
+      {
+        headers: { 'content-type': 'application/json' }
+      }
+    ).then(resp => {
+      swal(resp.data, null, 'success');
+      return dispatch(resetPasswordEmailSent());
+    });
+  };
+}
+function resetPasswordEmailSent() {
+  return {
+    type: 'RESET_PASSWORD_EMAIL_SENT'
+  };
+}
+export function logout() {
+  localStorage.clear();
+  browserHistory.replace('/dashboard');
   return {
-    type: USER_ID,
-    id,
+    type: LOGOUT_SUCCESS
+  };
+}
+export function isAuthenticated() {
+  let expiresAt = JSON.parse(localStorage.getItem('expires_at'));
+  return new Date().getTime() < expiresAt;
+}
+function getAccessToken() {
+  const accessToken = localStorage.getItem('access_token');
+  if (!accessToken) {
+    return null;
+  }
+  return accessToken;
+}
+function getUserMetadata(userId) {
```

Comment:
what is the difference between this method and the one in `auth.js`?

Reply:
deleted